roles/prometheus/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2024, Itential, Inc
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+---
+- name: Install base OS packages
+  ansible.builtin.include_role:
+    name: os
+  tags: install_base_os_packages
+
+- name: Create prometheus group
+  ansible.builtin.group:
+    name: "{{ prometheus_group }}"
+    state: present
+
+- name: Create prometheus user
+  ansible.builtin.user:
+    name: "{{ prometheus_user }}"
+    group: "{{ prometheus_group }}"
+    state: present
+
+- name: Create a directory if it does not exist
+  ansible.builtin.file:
+    path: /opt/prometheus
+    state: directory
+    owner: "{{ prometheus_user }}"
+    group: "{{ prometheus_group }}"
+    mode: "0755"
+
+# Install prometheus on its on server
+- name: Install Prometheus server
+  tags: prometheus_server
+  ansible.builtin.import_tasks:
+    file: install_prometheus.yml
+  when: inventory_hostname in groups['prometheus']
+
+# Install the node exporter on all servers
```

Comment:
Do we really need the node exporter on all servers?  It seems like we would only need it on the platform servers.

Reply:
This exporter is exporting basic system information, like sysadmin stuff, disk size, memory, network, etc. The name of the exporter just happens to be "node" it has no relation to NodeJS.  For a complete metrics solution I am adding this general exporter to ALL servers.  We could consider renaming it but I chose to retain the name so we know what exporter this is.  Its a very popular, well used and documented exporter.